File: 2022-2023/sourcecode/lab/lab11/VeryNobleAppJava/app/src/main/java/pl/udu/uwr/pum/verynobleappjava/adapters/nobelprizelaureates/LaureateAdapter.java
package pl.udu.uwr.pum.verynobleappjava.adapters.nobelprizelaureates;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;
import androidx.recyclerview.widget.ListAdapter;

import pl.udu.uwr.pum.verynobleappjava.data.NobelAwardsResponse;
import pl.udu.uwr.pum.verynobleappjava.databinding.LaureateRvItemBinding;

public class LaureateAdapter extends ListAdapter<NobelAwardsResponse.NobelPrizes.Laureates, LaureateViewHolder> {
    public LaureateAdapter(LaureateComparator itemComparator) {
        super(itemComparator);
    }

    @NonNull
    @Override
    public LaureateViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new LaureateViewHolder(LaureateRvItemBinding.inflate(
                LayoutInflater.from(parent.getContext()), parent, false
        ));
    }

    @Override
    public void onBindViewHolder(@NonNull LaureateViewHolder holder, int position) {
        NobelAwardsResponse.NobelPrizes.Laureates item = getItem(position);
        holder.bind(item);
    }
}

File: 2022-2023/sourcecode/lab/lab11/VeryNobleAppKotlin/app/src/main/java/pl/udu/uwr/pum/verynobleappkotlin/adapters/nobleprizelaureates/LaureateAdapter.kt
package pl.udu.uwr.pum.verynobleappkotlin.adapters.nobleprizelaureates

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.ListAdapter
import pl.udu.uwr.pum.verynobleappkotlin.data.nobelprizeresponse.laureate.Laureate
import pl.udu.uwr.pum.verynobleappkotlin.databinding.LaureateRvItemBinding

class LaureateAdapter(itemComparator: LaureateComparator) : ListAdapter<Laureate, LaureateViewHolder>(itemComparator) {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LaureateViewHolder {
        return LaureateViewHolder(
            LaureateRvItemBinding.inflate(
            LayoutInflater.from(parent.context), parent, false
        ))
    }

    override fun onBindViewHolder(holder: LaureateViewHolder, position: Int) {
        val item = getItem(position)
        holder.bind(item)
    }
}